### PR TITLE
Commerce 9766

### PR DESCRIPTION
**Pull request by andrea-ale-sbarra**
https://issues.liferay.com/browse/COMMERCE-9766

### CODE REVIEW

**Comment by andrea-ale-sbarra** [comment]
ci:test:sf

**Comment by andrea-ale-sbarra** [comment]
ci:test:relevant

File: modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CProductLocalServiceImpl.java
Comment (alessiorendina) [inline modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CProductLocalServiceImpl.java:70]: Use a new method into CPDefinitionLocalService in which you call `cpDefinitionPersistence.findByCProductId` and iterate through the list

File: modules/apps/commerce/commerce-product-definitions-web/src/main/java/com/liferay/commerce/product/definitions/web/internal/display/context/CPDefinitionsDisplayContext.java
Comment (alessiorendina) [inline modules/apps/commerce/commerce-product-definitions-web/src/main/java/com/liferay/commerce/product/definitions/web/internal/display/context/CPDefinitionsDisplayContext.java:535]: PortalException to be thrown by the method instead of doing a try-catch

**Review by alessiorendina — CHANGES_REQUESTED** [review]
@andrea-ale-sbarra Please check my comments and please reorder commits, thanks.

File: modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CProductLocalServiceImpl.java
Comment (andrea-ale-sbarra) [reply modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CProductLocalServiceImpl.java:70]:
```suggestion
		int cpDefinitionsCount = cpDefinitionPersistence.countByCProductId(
			cpDefinition.getCProductId());

		if (cpDefinitionsCount == 1) {
			cProductLocalService.deleteCProduct(cpDefinition.getCProductId());
		}
```

Comment (andrea-ale-sbarra) [reply modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CProductLocalServiceImpl.java:70]: `int cpDefinitionsCount = cpDefinitionPersistence.countByCProductId(
			cpDefinition.getCProductId());

		if (cpDefinitionsCount == 1) {
			cProductLocalService.deleteCProduct(cpDefinition.getCProductId());
		}`

Comment (andrea-ale-sbarra) [reply modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CProductLocalServiceImpl.java:70]:
```
@Override
	public void removeByCProductId(long CProductId) {
		for (CPDefinition cpDefinition :
				findByCProductId(
					CProductId, QueryUtil.ALL_POS, QueryUtil.ALL_POS, null)) {

			remove(cpDefinition);
		}
	}
```

Comment (andrea-ale-sbarra) [reply modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CProductLocalServiceImpl.java:70]: that's what the persistence is doing

**Comment by andrea-ale-sbarra** [comment]
ci:test:relevant

**Comment by andrea-ale-sbarra** [comment]
ci:test:sf

**Comment by andrea-ale-sbarra** [comment]
ci:test:relevant

**Comment by andrea-ale-sbarra** [comment]
ci:test:relevant

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/rest-openapi.yaml
Comment (alessiorendina) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/rest-openapi.yaml:5998]: Gets a product by external reference code.

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/rest-openapi.yaml
Comment (alessiorendina) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/rest-openapi.yaml:5922]: ERC

File: modules/apps/commerce/commerce-product-definitions-web/src/main/java/com/liferay/commerce/product/definitions/web/internal/display/context/CPDefinitionsDisplayContext.java
Comment (alessiorendina) [inline modules/apps/commerce/commerce-product-definitions-web/src/main/java/com/liferay/commerce/product/definitions/web/internal/display/context/CPDefinitionsDisplayContext.java:540]: `return cProductVersionConfiguration.enabled();`

File: modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CPDefinitionLocalServiceImpl.java
Comment (alessiorendina) [inline modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CPDefinitionLocalServiceImpl.java:969]: Check if there is an old approved version, if there is one, update the CProduct with the old approved CPDefinitionId

**Review by alessiorendina — CHANGES_REQUESTED** [review]
@andrea-ale-sbarra Please check my comments. Also, separate the commit that contains service and API changes, thanks!

**Comment by andrea-ale-sbarra** [comment]
ci:test:sf

**Comment by andrea-ale-sbarra** [comment]
ci:test:sf

**Comment by andrea-ale-sbarra** [comment]
ci:test:sf

**Comment by andrea-ale-sbarra** [comment]
ci:test:sf

**Comment by andrea-ale-sbarra** [comment]
ci:test:sf

**Comment by andrea-ale-sbarra** [comment]
ci:test:sf

**Comment by andrea-ale-sbarra** [comment]
ci:test:relevant

**Comment by andrea-ale-sbarra** [comment]
ci:test:sf

**Comment by andrea-ale-sbarra** [comment]
ci:test:relevant

**Comment by andrea-ale-sbarra** [comment]
ci:stop

**Comment by andrea-ale-sbarra** [comment]
ci:test:sf

**Comment by andrea-ale-sbarra** [comment]
ci:test:relevant

File: modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CPDefinitionLocalServiceImpl.java
Comment (alessiorendina) [inline modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CPDefinitionLocalServiceImpl.java:985]: `e` -> `cpDefinition` or `curCPDefinition`

**Review by alessiorendina — APPROVED** [review]
@andrea-ale-sbarra It seems ok to me now, we just need to understand if we want to send them together or separately. If we want to send them together, we need to merge the commits that are containing autogenerated files. Thanks!

**Comment by alessiorendina** [comment]
@iovaneale Let's wait for the PM decision before forwarding this one, thanks.

**Comment by pietrobocale** [comment]
ci:test:commerce

**Comment by alessiorendina** [comment]
ci:test:sf

**Comment by alessiorendina** [comment]
ci:test:relevant

**Pull request closed without merging**
